plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/TeamCommand.java

```diff
@@ -193,33 +112,38 @@ public void execute(ScriptEntry scriptEntry) {
                 team = board.registerNewTeam(name.asString());
             }
         }
-        if (add != null) {
-            for (String string : add) {
-                string = translateEntry(string, scriptEntry.context);
-                if (!team.hasEntry(string)) {
-                    team.addEntry(string);
+        if (removeEntities != null) {
+            for (ObjectTag obj : removeEntities.objectForms) {
+                String remove = translateEntry(obj, scriptEntry.context);
+                if (remove != null) {
+                    team.removeEntry(remove);
                 }
             }
         }
-        if (remove != null) {
-            for (String string : remove) {
-                string = translateEntry(string, scriptEntry.context);
-                if (team.hasEntry(string)) {
-                    team.removeEntry(string);
+        if (addEntities != null) {
+            for (ObjectTag obj : addEntities.objectForms) {
+                String add = translateEntry(obj, scriptEntry.context);
+                if (add != null) {
+                    team.addEntry(add);
                 }
             }
         }
         if (option != null) {
-            String optName = option.asLowerString();
-            String statusName = status.asLowerString();
-            if (optName.equals("friendly_fire")) {
-                team.setAllowFriendlyFire(statusName.equals("always"));
-            }
-            else if (optName.equals("see_invisible")) {
-                team.setCanSeeFriendlyInvisibles(statusName.equals("always"));
-            }
-            else {
-                team.setOption(Team.Option.valueOf(optName.toUpperCase()), Team.OptionStatus.valueOf(statusName.toUpperCase()));
+            switch (option.asString().toLowerCase()) {
+                case "friendly_fire" -> {
+                    team.setAllowFriendlyFire(status.toString().equalsIgnoreCase("always"));

```

Comment:
never use `equalsIgnoreCase`, use `asLowerString().equals`

Reply:
should I convert it to elementtag then?

Reply:
... wait, that's an enum? use the enum checks then not strings.